src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -41,6 +41,165 @@ using namespace chip::app::Clusters;
+        OOPairs[0].attributeValue = currentValue;
+        OOattributeValueList      = OOPairs;
+
+        TLV::TLVWriter writer;
+        TLV::TLVType outer;
+        // Serialize Extension Field sets in a way consistent with the default Deserialize method
+        writer.Init(serialisedBytes);
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outer));
+        ReturnErrorOnFailure(app::DataModel::Encode(
+            writer, TLV::ContextTag(to_underlying(Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList)),
+            OOattributeValueList));
+        ReturnErrorOnFailure(writer.EndContainer(outer));
+        serialisedBytes.reduce_size(writer.GetLengthWritten());
+
+        return CHIP_NO_ERROR;
+    }
+
+    /// @brief Default EFS interaction when applying scene to the OnOff Cluster
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serialisedBytes Data from nvm
+    /// @param timeMs transition time in ms
+    /// @return CHIP_NO_ERROR if value as expected, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR ApplyScene(EndpointId endpoint, ClusterId cluster, const ByteSpan & serialisedBytes,
+                          scenes::TransitionTimeMs timeMs) override
+    {
+        app::DataModel::DecodableList<Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
+        Scenes::Structs::AttributeValuePair::DecodableType decodePair;
+
+        TLV::TLVReader reader;
+        TLV::TLVType outer;
+
+        size_t attributeCount = 0;
+
+        VerifyOrReturnError(cluster == Id, CHIP_ERROR_INVALID_ARGUMENT);
+
+        reader.Init(serialisedBytes);
+        ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+        ReturnErrorOnFailure(reader.EnterContainer(outer));
+        ReturnErrorOnFailure(reader.Next(
+            TLV::kTLVType_Array, TLV::ContextTag(app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList)));
+        ReturnErrorOnFailure(attributeValueList.Decode(reader));
+
+        ReturnErrorOnFailure(attributeValueList.ComputeSize(&attributeCount));
+        VerifyOrReturnError(attributeCount <= scenableAttributeCount, CHIP_ERROR_BUFFER_TOO_SMALL);
+        auto pair_iterator = attributeValueList.begin();
+
+        // Assumes the OnOff cluster has only one Attribute
+        pair_iterator.Next();
+
+        decodePair = pair_iterator.GetValue();
+
+        if (decodePair.attributeID.HasValue())
+        {
+            // If attribute ID was encoded, verify it is the proper ID for the OnOff attribute
+            VerifyOrReturnError(decodePair.attributeID.Value() == Attributes::OnOff::Id, CHIP_ERROR_INVALID_ARGUMENT);
+        }
+
+        mSceneOnOffState = decodePair.attributeValue;
+
+        // Verify that the EFS was completely read
+        ReturnErrorOnFailure(pair_iterator.GetStatus());
```

Comment:
This needs to be _before_ we use GetValue(), effectively.  So as soon as Next() returns false.